oracle.py

```diff
@@ -179,7 +181,25 @@ def get_crd(crd):
     return data
 
 
-def generate_resources(path=""):
+def learn_twice_trim(base_resources, twice_resources):
+    def nested_set(dic, keys, value):
+        for key in keys[:-1]:
+            dic = dic[key]
+        dic[keys[-1]] = value
+
+
+    stored_learn = copy.deepcopy(base_resources)
+    ddiff = DeepDiff(twice_resources, base_resources, ignore_order=False, view='tree')
+
+    for key in ddiff['values_changed']:
+        nested_set(stored_learn, key.path(output_format='list'), "SIEVE-IGNORE")
+
+    for key in ddiff['dictionary_item_added']:
+        nested_set(stored_learn, key.path(output_format='list'), "SIEVE-IGNORE")
+
+    return stored_learn
+
+def generate_resources(path = "", data_dir="", learn_twice=False):
```

Comment:
I would suggest you separate the functionality of generating resources and trimming resources. The previous generate_resources can remain unchanged, and you can create another function for trimming only
```
trim_resources(first_resources, second_resources)
```
The trim_resources should only be called in the second learn run